File: Practice-questions/reversearray.java
public void swap(int[] arr,int a,int b)
{
  
int temp=arr[a];
arr[a]=arr[b];
arr[b]=temp;

}
public int[] reverseArray(int[] arr){
int size=arr.length-1;

for(int i=0;i<size;i++){

swap(arr,i,size--);

}

return arr;
}

File: Revision/Recursion/ReverseArray.cpp
#include<iostream>
#include<climits>
using namespace std;

void reverse(int arr[], int start, int end){
    if(start>=end){
        return;
    }
    swap(arr[start], arr[end]);
    reverse(arr, start+1, end-1);
}

int main()
{
    int arr[] = {1, 2, 3, 4, 5};
    reverse(arr, 0, 4);
    for (int i = 0; i < 5; i++)
    {
        cout<<arr[i]<<" ";
    }
    
    return 0;
}